### PR TITLE
refactor(footer): use screen size controller

### LINKED ISSUE #513: [refactor]: footer use ScreenSizeController
- rh-footer should replace MatchMediaController with ScreenSizeController
- ScreenSizeController shold import tokens from tokens package

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Updated footer to use `ScreenSizeController`

What this doesn't do, is use updated tokens imported from the token repo.  That will be in a later PR.

Closes #513

## Testing Instructions

1. View [deploy preview](https://deploy-preview-807--red-hat-design-system.netlify.app/elements/footer/demo/) to ensure mobile state works to swap between accordion and lists views

### CODE REVIEW

File: elements/rh-footer/RhFooter.ts [hunk 1]
```diff
@@ -70,23 +68,32 @@ export class RhFooter extends LitElement {
...
-
   #logger = new Logger(this);
 
   @colorContextProvider()
   @property({ reflect: true, attribute: 'color-palette' }) colorPalette: ColorPalette = 'darker';
 
   override connectedCallback() {
     super.connectedCallback();
-    // wire up accessbility aria-labels with unordered lists
+    // wire up accessability aria-labels with unordered lists
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply elements/rh-footer/RhFooter.ts:78]: I was just committing to fix that

File: .changeset/sour-chefs-run.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@rhds/elements": patch
+---
+
+`<rh-footer>`: switched to use ScreenSizeControlller
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply .changeset/sour-chefs-run.md:5]: I believe so.  I don't see it used anywhere else.

File: elements/rh-footer/RhFooter.ts [hunk 3]
```diff
@@ -8,10 +8,8 @@ import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
 import style from './rh-footer.css';
 import { responsiveStyles } from './rh-footer-responsive.css.js';
 
-import { tabletLandscapeBreakpoint } from '../../lib/tokens.js';
 import { colorContextProvider, type ColorPalette } from '../../lib/context/color/provider.js';
-// TODO: use ScreenSizeController
-import { MatchMediaController } from '../../lib/MatchMediaController.js';
```
[the remark this answers is not in the document]
Comment (zeroedin) [reply elements/rh-footer/RhFooter.ts:14]: I believe so.  I don't see it used anywhere else.

**Review by bennypowers — APPROVED** [review]
Code looks good. after manual qa on the dp (which I haven't done yet), this can go in

**Pull request merged**